internal/util/diff/diff_test.go

```diff
@@ -242,6 +254,115 @@ func TestCommand_Diff(t *testing.T) {
 	}
 }
 
+func TestCommand_InvalidRef(t *testing.T) {
+	reposChanges := map[string][]testutil.Content{
+		testutil.Upstream: {
+			{
+				Data:   testutil.Dataset2,
+				Branch: "master",
+				Tag:    "v2",
+			},
+			{
+				Data: testutil.Dataset3,
+			},
+		},
+	}
+
+	g := &testutil.TestSetupManager{
+		T:            t,
+		ReposChanges: reposChanges,
+		GetRef:       "v2",
+	}
+	defer g.Clean()
+
+	if !g.Init() {
+		return
+	}
+
+	diffOutput := &bytes.Buffer{}
+	err := (&Command{
+		Path:         g.LocalWorkspace.FullPackagePath(),
+		Ref:          "hurdygurdy", // ref should not exist in upstream
+		DiffType:     DiffTypeCombined,
+		DiffTool:     "diff",
+		DiffToolOpts: "-r -i -w",
+		Output:       diffOutput,
+	}).Run(fake.CtxWithNilPrinter())
+	assert.Error(t, err)
```

Comment:
Can we do some kind of verification here that we actually got the error we expected?